From 1f8a1cef36fc0ea3efa344b93d07e65fbedc0631 Mon Sep 17 00:00:00 2001
From: zsy-jason <186247581+zsy-jason@users.noreply.github.com>
Date: Mon, 30 Mar 2026 14:08:36 +0800
Subject: [PATCH] feat: change abi to 1.

---
 third_party/libc++/__config_site | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/libc++/__config_site b/third_party/libc++/__config_site
index 2df8632..119a25d 100644
--- a/third_party/libc++/__config_site
+++ b/third_party/libc++/__config_site
@@ -12,7 +12,7 @@
 //    limit for PDBs (https://crbug.com/1327710#c5). To fix this, we set
 //    _LIBCPP_ABI_NAMESPACE to a shorter value.
 #define _LIBCPP_ABI_NAMESPACE Cr
-#define _LIBCPP_ABI_VERSION 2
+#define _LIBCPP_ABI_VERSION 1
 
 /* #undef _LIBCPP_ABI_FORCE_ITANIUM */
 /* #undef _LIBCPP_ABI_FORCE_MICROSOFT */
-- 
2.51.1.windows.1

